DoLah/Parser/DateTimeParser.cpp

```diff
@@ -56,44 +58,93 @@ namespace DoLah {
         std::string out;
         for (size_t m = 0; m < monthPattern.size(); m++) {
             if (std::regex_match(str, std::regex(monthPattern.at(m), std::regex_constants::icase))) {
-                return (int) m;
+                return (int)m;
             }
         }
 
-        return NULL;
+        return REJECT;
     }
 
     int DateTimeParser::getYear(std::string str) {
-        if (str.length() != 2 && str.length() != 4) {
-            return NULL;
-        } else if (!ParserLibrary::isDecimal(str)) {
-            return NULL;
+        std::string year = str;
```

Comment:
Why don't you just change the parameter value? instead of having `str` since it only as a temporary variable?